File: parnamondal/Week - 2/Day-3/Minimum size subarray sum.py
# Time Complexity - O(Nlog N)
# Space Complexity - O(1)

class Solution:
    def minSubArrayLen(self, s: int, nums: List[int]) -> int:
        n=len(nums)
        if n==0 :
            return 0
        res=float('inf')
        curr=0
        left=0
        for i in range (n):
            curr+=nums[i]
            while (curr>=s):
                res=min(res,i+1-left)
                curr-=nums[left]
                left+=1
        if res!=float('inf'):
            return res
        return 0

                
        

File: Adhiti_Shaili/Day 10/Minimum Size Subarray Sum.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int mini=INT_MAX,sum=0;
        int n=nums.size();
        int i=0,j=0;
        while(i<=j&&j<n)
        {
            if(sum>=s)
            {
                mini=min(mini,j-i);
                sum-=nums[i];
                i++;
            }
            else
            {
                sum+=nums[j];
                j++;
            }
        }
        while(i<n)
        {
            if(sum>=s)
            {
                mini=min(mini,j-i);
            }
            sum-=nums[i];
            i++;
        }
        if(mini==INT_MAX)
        {
            return 0;
        }
        else
        {
            return mini;
        }
        
    }
};